great_expectations/cli/cli.py

```diff
@@ -250,8 +247,11 @@ def profile(datasource_name, data_assets, profile_all_data_assets, directory):
 
     if datasource_name is None:
         datasources = [datasource["name"] for datasource in context.list_datasources()]
-        for datasource_name in datasources:
-            profile_datasource(context, datasource_name, max_data_assets=max_data_assets)
+        if len(datasources) > 1:
```

Comment:
I can't quite tell how this is supposed to work with other variables.

Unit tests as examples would be **super helpful** here.

Reply:
@abegong Added tests with/without/with valid/invalid args and a method doc.